src/gh/diffCheck/diffCheck/df_error_estimation.py

```diff
@@ -0,0 +1,156 @@
+
+    if signed:
+        # build an RTree containing all the points of the target
+        tree = rg.RTree()
+        for i, ver in enumerate(target.points): 
+            tree.Insert(rg.Point3d(ver[0], ver[1], ver[2]), i)
+
+        for idx, p in enumerate(source.Vertices):
+
+            # find the index on the target that the vertex is closest to
+            search_point = p
+            sphere = rg.Sphere(search_point, distances[idx]*1.0001) #to change later, hack to avoid not finding the point
+            found_indices = []
+
+            def search_callback(sender, e):
+                found_indices.append(e.Id)
+
+            tree.Search(sphere, search_callback)
+
+            df_closest_point = target.points[found_indices[0]]
+            closest_point = rg.Point3d(df_closest_point[0], df_closest_point[1], df_closest_point[2])
+            # Calculate the direction from target to source
+            direction = p - closest_point
+            # Calculate the signed distance
+            normal = source.Normals[idx]
+            dot_product = direction * normal
+            if dot_product < 0:
+                distances[idx] = - distances[idx]
+
+    return np.asarray(distances)
+
+
+def cloud_2_rhino_mesh_distance(source, target, signed=False):
+    """
+        Calculate the distance between every point of a source pcd to its closest point on a target Rhino Mesh
+    """
+
+    #for every point on the point cloud find distance to mesh
+    distances = []
+
+    for p in source.points:
+
+        rhp = rg.Point3d(p[0], p[1], p[2])
+        closest_meshPoint = target.ClosestMeshPoint(rhp, 1000)
+        closest_point = closest_meshPoint.Point
+        distance = rhp.DistanceTo(closest_point)
+
+        if signed:
+            # Calculate the direction from target to source
+            direction = rhp - closest_point
+            # Calculate the signed distance
+            normal = target.NormalAt(closest_meshPoint)
+            dot_product = direction * normal
+            if dot_product < 0:
+                distance = -distance
+
+        distances.append(distance)
+
+    return np.asarray(distances)
+
+
+class DFVizResults:
+    """
+    This class compiles the resluts of the error estimation into one object
+    """
+
+    def __init__(self):
+
+        self.source = []
+        self.target = []
+
+        self.distances_mse = []
+        self.distances_max_deviation = []
+        self.distances_min_deviation = []
+        self.distances_sd_deviation = []
+        self.distances = []
+
+    def add(self, source, target, distances):
+
+        self.source.append(source)
+        self.target.append(target)
+
+        self.distances_mse.append(np.sqrt(np.mean(distances ** 2)))
+        self.distances_max_deviation.append(np.max(distances))
+        self.distances_min_deviation.append(np.min(distances))
+        self.distances_sd_deviation.append(np.std(distances))
+        self.distances.append(distances.tolist())
```

Comment:
Just wondering, why not have a constructor that takes the source, target, and distance directly ?

Reply:
agreed. Be a more defensive and locking up on this object might be good. We could have a construction as @DamienGilliard proposedand class properties to retrieve values.

Reply:
it seemed a bit cleaner way for me to populate the source, target, and distances list. The latter is also a nested list. But if both of you agree, I can change

Reply:
Than ok with your solution @eleniv3d !